### DISCUSSION TITLE
Prev / next reference of element under cursor in file

### DISCUSSION THREAD

**Discussion by eulke** [zed-industries/zed#14703] [Feature Requests]
Is there a way to move between references of an element under the cursor within a file in Zed? Something like vim-illuminate

**Comment by SomeoneToIgnore** [comment]
Only `editor: select next` and `editor: select previous` exists to help with this, as it seems: either of the commands will move to another match to the word under caret, but also will select this match.

**Reply by eulke** [reply]
Sure, I understand. In this case I was looking for an option linked to the LSP or Treesitter.
For example, when standing on a variable named “auto” I would like to go to the different references or uses of that specific variable within the file, not to a different variable that includes “auto” in its name. That is the behavior of vim-illuminate.
Anyway, thank you very much for the answer :)

**Comment by MarcoPolo** [comment]
Related to https://github.com/zed-industries/zed/discussions/21193

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/community-links
